include/power_grid_model/component/fault.hpp

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_COMPONENT_SHORT_CIRCUIT_HPP
+#define POWER_GRID_MODEL_COMPONENT_SHORT_CIRCUIT_HPP
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "base.hpp"
+
+namespace power_grid_model {
+
+class Fault final : public Base {
+   public:
+    using InputType = FaultInput;
+    using UpdateType = FaultUpdate;
+    template <bool sym>
+    using OutputType = FaultOutput;
+    template <bool sym>
+    using ShortCircuitOutputType = FaultShortCircuitOutput<sym>;
+    static constexpr char const* name = "fault";
+    ComponentType math_model_type() const final {
+        return ComponentType::fault;
+    }
+
+    Fault(FaultInput const& fault_input)
+        : Base{fault_input},
+          fault_object_{fault_input.fault_object},
+          r_f_{is_nan(fault_input.r_f) ? (double)0.0 : fault_input.r_f},
+          x_f_{is_nan(fault_input.x_f) ? (double)0.0 : fault_input.x_f} {
+    }
+
+    FaultCalcParam calc_param(double const& u_rated, bool const& is_connected_to_source = true) const {
+        if (!energized(is_connected_to_source)) {
+            return FaultCalcParam{};
+        }
+        // param object
+        FaultCalcParam param{};
```

Comment:
If the fault impedance is zero, which is quite common, you still calculate `base_y`. We can improve the performance here by doing two separate returns. 

Also, try not to define separate variables for each return path. This will prevent the Return Value Optimization.

Suggestion:


```c++
    FaultCalcParam calc_param(double const& u_rated, bool const& is_connected_to_source = true) const {
        // param object
        FaultCalcParam param{};  // only declare one object to be returned, this enables RVO.
        if (!energized(is_connected_to_source)) {
            return param;
        }
        if (r_f_ == 0.0 && x_f_ == 0.0) {
           param.y_fault.real(std::numeric_limits<double>::infinity());
           param.y_fault.imag(std::numeric_limits<double>::infinity());
           return param;
        }
        double const base_z = u_rated * u_rated / base_power_3p;
        param.y_fault = base_z  / (r_f_ + 1.0i * x_f_);
        return param;
    }
```